File: Back/src/main/java/com/example/back/Services/GrilleService.java
package com.example.back.Services;

import com.example.back.Entities.Grille;
import com.example.back.Entities.Tache_journal;

import java.util.List;

public interface GrilleService {

    Grille addGrille(Grille grille);

    Grille updateGrille(Long idGrille, Grille grille) ;

    Grille findGrilleById(Long idGrille);


    List<Grille> findAll();

    void removeGrille (Long idGrille);
}

File: Front/src/app/Services/GrilleService/grille.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Grille} from "../../Modules/GrilleModule/Grille.module";

@Injectable({
  providedIn: 'root'
})
export class GrilleService {

  private apiUrl = 'http://localhost:9090/api/services/grille';
  constructor(private http: HttpClient) { }

  findAllGrilles(): Observable<Grille[]> {
    return this.http.get<Grille[]>(`${this.apiUrl}/GetAllGrille`);
  }

  addGrille(grille: Grille): Observable<Grille> {
    return this.http.post<Grille>(this.apiUrl, grille);
  }

  updateGrille(idGrille: number, updtgrille: Grille): Observable<Grille> {
    return this.http.put<Grille>(`${this.apiUrl}/updateGrille/${idGrille}`, updtgrille);
  }

  deleteGrille(idGrille: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/removeGrille/${idGrille}`);
  }


}
